tensornetwork/network_operations.py

```diff
@@ -590,21 +590,25 @@ def _reachable(nodes: Set[BaseNode]) -> Set[BaseNode]:
   return seen_nodes
 
 
-def reachable(nodes: Union[BaseNode, Iterable[BaseNode]]) -> Set[BaseNode]:
+def reachable(inputs: Union[BaseNode, Iterable[BaseNode], Edge, Iterable[Edge]]) -> Set[BaseNode]:
   """
-  Computes all nodes reachable from `node` by connected edges.
+  Computes all nodes reachable from `node` or `edge.node1` by connected edges.
   Args:
-    nodes: A `BaseNode` or collection of `BaseNodes`
+    inputs: A `BaseNode`/`Edge` or collection of `BaseNodes`/`Edges`
   Returns:
     A list of `BaseNode` objects that can be reached from `node`
     via connected edges.
   Raises:
     ValueError: If an unknown value for `strategy` is passed.
   """
 
-  if isinstance(nodes, BaseNode):
-    nodes = {nodes}
-  return _reachable(set(nodes))
+  if isinstance(inputs, BaseNode):
+    inputs = {inputs}
+  elif isinstance(inputs, Edge):
+    inputs = {inputs.node1}
+  elif isinstance(inputs, list) and all(isinstance(_, Edge) for _ in inputs):
```

Comment:
Can you change the `_` to `x`? Generally `_` is only used when we want to ignore the variable, but in this case we are actually using it.

Reply:
Same just below.